src/test/java/com/microsoft/azure/batch/TaskTests.java

```diff
@@ -337,4 +340,47 @@ public void testGetTaskCounts() throws Exception {
             }
         }
     }
+
+    @Test
+    public void failCreateContainerTaskWithRegularPool() throws Exception {
+        String jobId = getStringWithUserNamePrefix("-Job-" + (new Date()).toString().replace(' ', '-').replace(':', '-').replace('.', '-'));
+        String taskId = "mytask";
+
+        PoolInformation poolInfo = new PoolInformation();
+        poolInfo.withPoolId(liveIaaSPool.id());
+        batchClient.jobOperations().createJob(jobId, poolInfo);
+
+        TaskAddParameter taskToAdd = new TaskAddParameter();
+        taskToAdd.withId(taskId)
+                .withContainerSettings(new TaskContainerSettings()
+                        .withContainerRunOptions("--rm")
+                        .withImageName("centos"))
+                .withCommandLine("bash -c \"echo hello\"");
+
+        try
+        {
+            batchClient.taskOperations().createTask(jobId, taskToAdd);
```

Comment:
Are the tests run against live? I think I tried this in one of my live tests and it didn't get a server exception....
Do the tests have recordings?

Reply:
You need re-create the pool to get latest node agent update.